odoo/addons/base/models/ir_cron.py

```diff
@@ -382,34 +467,31 @@ def _callback(self, cron_name, server_action_id, job_id):
 
             log_depth = (None if _logger.isEnabledFor(logging.DEBUG) else 1)
             odoo.netsvc.log(_logger, logging.DEBUG, 'cron.object.execute', (self._cr.dbname, self._uid, '*', cron_name, server_action_id), depth=log_depth)
-            _logger.info('Starting job `%s`.', cron_name)
+            _logger.info('Job `%s` (%s): Starting.', cron_name, job_id)
             start_time = time.time()
-            self.env['ir.actions.server'].browse(server_action_id).run()
+            result = self.env['ir.actions.server'].browse(server_action_id).run()
             end_time = time.time()
-            _logger.info('Job done: `%s` (%.3fs).', cron_name, end_time - start_time)
+            _logger.info('Job `%s` (%s): Done in (%.3fs).', cron_name, job_id, end_time - start_time)
             if start_time and _logger.isEnabledFor(logging.DEBUG):
-                _logger.debug('%.3fs (cron %s, server action %d with uid %d)', end_time - start_time, cron_name, server_action_id, self.env.uid)
+                _logger.debug('Job `%s` (%s): Server action #%d with uid %d executed in (%.3fs).',
+                              cron_name, job_id, server_action_id, self.env.uid, end_time - start_time)
             self.pool.signal_changes()
-        except Exception as e:
+            if not result or not isinstance(result, Sequence):
+                return (0, None)
```

Comment:
I still think that it would be nice to write a warning when it doesn't return a valid progress. Maybe only when it is not `None` (for those server actions that call `_notify_progress()`)
```suggestion
            if not result or not isinstance(result, Sequence):
                _logger.warning(...)
                return (0, None)
```